proteuslib/unit_models/reverse_osmosis_0D.py

```diff
@@ -956,17 +968,15 @@ def initialize(blk, initialize_guess=None, state_args=None, outlvl=idaeslog.NOTS
 
         # Initialize feed inlet state block
         flags = blk.feed_side.properties_in.initialize(
-            outlvl=outlvl,
+            outlvl=outlvl_init,
             optarg=optarg,
             solver=solver,
             state_args=state_args,
             hold_state=True)
 
         init_log.info_high("Initialization Step 1 Complete.")
-        if degrees_of_freedom(blk) != 0:
-            raise Exception(f"Initialization was called on {blk} "
-                            f"but it had {degrees_of_freedom(blk)} degree(s) of freedom "
-                            f"when 0 was expected. Check that the appropriate variables are fixed.")
+        if ignore_dof is False:
```

Comment:
Minor note for the future: it is considered more Pythonic to write `if not ignore_dof:` in cases like this.